rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,640 @@
+            "Actions": [
+				"rds:DescribeDBClusters",
+				"rds:DescribeDBInstances"
+	        ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+This AWS Role must trust Teleport as an IdP.
+To do so, the user must add the following Trusted relationship:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::0123456789012:oidc-provider/proxy.example.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "proxy.example.com:aud": "discover.teleport"
+                }
+            }
+        }
+    ]
+}
+```
+
+### UX
+
+#### Web flow
+Described [above](#user_s-point-of-view).
+
+#### Web API
+Integration resource will have the usual CRUD operations via Web API.
+
+The user can only update two fields:
+- awsOIDC.roleARN: to set the AWS Role that should be used for this Integration
+- status:
+	- Paused: when the Integration should not be used, to support a disable operation
+	- Running: when the Integration is running
+
+HTTP API:
+```
+Methods:
+GET .../integration
+GET .../integration/:id
+POST .../integration
+PUT .../integration/:id
+DELETE .../integration/:id
+```
+JSON representation:
+```json
+{
+	"name": "myaws",
+	"subkind": "aws-oidc",
+	"awsOIDC": {
+		"roleARN": "arn:aws:123:TeleportOIDC"
+	},
+	"status": "IntegrationStatusRunning"
+}
+```
+
+In order to call an Integration action, we'll create a new Web API endpoint:
+```
+--->
+POST .../integration/:id/action/aws-oidc-list-databases
+{}
+
+<---
+{
+    "status": "success",
+    "response": {
+        "items": [
+            {
+                "status": "available",
+                "name": "marcodbtest",
+                "iamAuth": "true",
+                "engine": "postgres",
+                "engineVersion": "15.2",
+                "masterUsername": "postgres",
+                "tags": [],
+                "arn": "arn:aws:rds:us-east-1:<accid>:db:marcodbtest",
+                "addr": "marcodbtest.<someid>.us-east-1.rds.amazonaws.com",
+                "port": "5432"
+            }
+        ]
+    }
+}
+```
+
+This will generate a new token for the Integration `:id` and then call the `aws-oidc-list-databases` method which maps to a method that will do two AWS API Calls:
+- `rds describe-db-instances`
+- `rds describe-db-clusters`
+
+The response contains the following fields:
+- `status`: reports the result of calling the integration (other errors will come as usual: http status code)
+- `response`: specific to the requested action
+
+Pagination/Offset and search capabilities are out of scope for now.
+
+#### CLI
+Users can also create/update the resource using `tctl`.
+
+The only updatable field is `role_arn`.
+
+`tctl`:
+
+```
+$ tctl get integrations
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: some-name
```

Comment:
Here and everywhere below in yaml resources indentation seems off. Use spaces instead of tabs?

Reply:
:thinking: I think it was correct but using 4 spaces instead of the usual (?) 2
I also added the proto representation of the Integration resource